Fixes text appearing over scrollbar in ItemList

AlbertFay opened the pull request:
Fixes part of the problem in #57881 
This fixes the issue of text appearing over the vertical scrollbar when using tileset editor. The issue was in `ItemList`, the width was not being set correctly.

Here is a working preview:
![working preview of itemlist](https://user-images.githubusercontent.com/65504787/153779534-41ced211-0052-4f17-b011-bdddfafc71cf.gif)

Calinou committed:
Add an `OS.get_processor_name()` method

This method can be used to get the CPU model name.
It can be used in conjunction with
`RenderingServer.get_video_adapter_name()` and
`RenderingServer.get_video_adapter_vendor()` for annotating benchmarks
and automatic graphics quality configuration.

clayjohn committed:
Use Filament specular models and parametrization

EricEzaM committed:
Fixed issues with shortcut checking in 3d node editor

The freelook shortcuts now use the action system by proxy. This allows the actions system to be leveraged for input checking. When the shortcut changes, the respective action is updated.

BastiaanOlij committed:
Add OpenXR 1.0.22 to thirdparty libraries

Will be compiled and used in the next commit.

Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>

BastiaanOlij committed:
Implementing OpenXR driver

Diddykonga committed:
Change 'find_node' to 'find_nodes' and Add 'type' parameter

Changed 'find_node' to 'find_nodes' which now returns an 'TypedArray<Node>', as well as Added a 'type' parameter to match against specific node types, which supports inheritance.

jasonwinterpixel committed:
Allow disabling the alpha trim on texture atlas creation.

Default to true

trollodel committed:
Remove most EditorNode constructor parameters and fields

floppyhammer committed:
Fix GPUParticles2D emission offset in global coords

TechnicalSoup committed:
Cleanup of preprocessor directives in main.cpp

Remove redundant set of preprocessor directives and combine duplicate if statements.

Faless committed:
[OS/Crypto] Add get_entropy to OS.

Implemented via `BCryptGenRandom` on Windows.
Implemented via `getentropy` syscall when available.
Implemented via `/dev/urandom` device as a fallback.

The `/dev/urandom` fallback can be disabled via the `NO_URANDOM` build
flag.

Note: The HTML5 version relies on emscripten file system urandom
device which itself uses the Crypto API when available or the plain
old not crypto-safe `Math.random()` otherwise.

Restore get_entropy.

Faless committed:
[ResourceUID] Use CryptoCore::RandomGenerator for IDs.

Faless committed:
[Crypto] Implement CryptoCore::RandomGenerator.

As a cryptographically secure random generator.

Internally it uses mbedTLS CTR-DRBG implementation which gets re-seeded
with entropy from OS::get_entropy when needed.

CryptoCore now additionally depends on `ctr_drbg.c` and `entropy.c`
thirdparty mbedtls files.

Faless committed:
[OSX] Codesign exporter now uses CryptoCore RNG.

Eoin-ONeill-Yokai committed:
Add missing binding to AudioStream class

This allows for the extension of AudioStream where you can call
`instance_plackback` on child AudioStream instances within gdscript,
much like the implementation of some child classes in C++.

See `AudioStreamRandomPitch` for an example of how this can be used.

kleonc committed:
NavMap Fix polygons being treated like triangle strips instead of triangle fans

AlbertFay committed:
Fixed bug in ItemList

AlbertFay committed:
Added Ellipsis

KoBeWi commented on scene/gui/item_list.cpp:
```diff
@@ -1237,8 +1237,8 @@ void ItemList::_notification(int p_what) {
 					}
-
-					items.write[i].text_buf->set_width(max_len);
+					
```
```suggestion

```
Trailing whitespace.

AlbertFay committed:
Fixed whitespace

AlbertFay committed:
Update scene/gui/item_list.cpp

Co-authored-by: Tomek <kobewi4e@gmail.com>

AlbertFay committed:
Merge branch 'text-under-scrollbar' of https://github.com/AlbertFay/godot into text-under-scrollbar

AlbertFay committed:
Added space

AlbertFay committed:
Merge branch 'text-under-scrollbar' of https://github.com/AlbertFay/godot into text-under-scrollbar

Calinou committed:
Only store `_edit_use_anchors_` metadata if value is not the default

The default value is assumed to be `false`, so this metadata
only needs to be stored if the value is `true`.

fire-forge committed:
Fix EditorProperty text and icon overlap bugs

YeldhamDev committed:
Fix crash when removing tabs from `TabBar`

timothyqiu committed:
Make duplicate animation prompt for new name

akien-mga committed:
Merge pull request #57985 from Pineapple/pitchshift-tweaks

PitchShift effect quality and performance tweaks for different pitch scale values

akien-mga committed:
Merge pull request #57611 from TechnicalSoup/TechnicalSoup-patch-2

Cleanup of preprocessor directives in main.cpp

tavurth committed:
Fix new performance regressions (short delay_usec)

My Mac was using 20% cpu again, which was related to the Javascript
Export plugin.

I had however no export templates setup in the project so this is more
of a stopgap fix.

akien-mga committed:
Merge pull request #57989 from RandomShaper/update_vma

Update & patch VMA, and re-implement the small buffers optimization

akien-mga committed:
Merge pull request #58079 from timothyqiu/anim-dup-name

Make duplicate animation prompt for new name

akien-mga committed:
Merge pull request #58053 from kleonc/navmap-polygons-are-triangle-fans-not-triangle-strips

`NavMap` Fix polygons being treated like triangle strips instead of triangle fans

JFonS committed:
Merge pull request #57990 from jmb462/fix-drag-lines-precision-issue

Fix 3D editor axis drag lines precision issue

JFonS committed:
Merge pull request #57964 from mbrlabs/deselect

Deselect nodes in the 3D editor when pressing ESC

akien-mga committed:
Merge pull request #57887 from Faless/crypto/4.x_os_get_entropy

reduz committed:
Add Particle Shader Userdata

* Adds optional vec4 USERDATA1 .. USERDATA6 to particles, allowing to store custom data.
* This data is allocated on demand, so shaders that do not use it do not cost more.

akien-mga committed:
Editor: Cleanup some includes dependencies

Removes some unnecessary includes from `editor_node.h`, and instead add
those where they're used.

Removes unnecessary `editor_node.h` includes in various editor classes.

Renames `dynamicfont` to `dynamic_font` in a couple files.

Misc cleanup while jumping through that rabbit hole.

Calinou committed:
Add an XML schema for documentation

This makes it easier to spot syntax errors when editing the
class reference. The schema is referenced locally so validation
can still work offline.

Each class XML's schema conformance is also checked on GitHub Actions.

YuriSizov committed:
Add missing CheckButton icons

Calinou committed:
Improve completion scroll bar visibility in the script editor

This makes the scroll bar bar thicker and more opaque (roughly
matching the editor theme's scroll bar by default).

akien-mga committed:
Merge pull request #58089 from YeldhamDev/crashy_tabs

timothyqiu committed:
Validate PhysicsDirectSpaceState{2,3}D::_intersect_point input

akien-mga committed:
Merge pull request #57988 from markdibarry/add_get_last_visible_character_line

akien-mga committed:
Merge pull request #58090 from pycbouh/editor-missing-check-icons

akien-mga committed:
Merge pull request #58008 from Geometror/increase-checkerboard-contrast

akien-mga committed:
Merge pull request #57306 from trollodel/remove_editornode_param

akien-mga committed:
Merge pull request #58095 from timothyqiu/intersect-point

Calinou committed:
Clamp environment light sky contribution to the [0.0; 1.0] range

The value is already clamped in the editor, but it wasn't being
clamped when the value was set via code. Values outside the [0.0; 1.0]
range can result in broken rendering.

timothyqiu committed:
Fix crash and memory leak when importing OGG Vorbis

trollodel committed:
Remove the EditorNode parameter from EditorPlugins create methods

Remove EditorNode usage from the Navigation editor plugin.

Calinou committed:
Improve documentation for AudioServer device properties and methods

Calinou committed:
Expose `AudioServer.capture_device` as a property

This is more consistent with `AudioServer.device` (for output),
which is already exposed as a property.

akien-mga committed:
Merge pull request #58091 from Calinou/code-edit-improve-completion-scroll-bar-visibility

akien-mga committed:
Merge pull request #58103 from timothyqiu/ogg-vorbis

Commit:
Fixed variant decoding Segmentation Fault

akien-mga committed:
Merge pull request #58105 from trollodel/nuke_p_editor_editor_plugin

timothyqiu committed:
Fix infinite recursion when seeking Ogg Vorbis file

akien-mga committed:
Merge pull request #58100 from Calinou/environment-clamp-sky-contribution

Clamp environment light sky contribution to the [0.0; 1.0] range

akien-mga committed:
Merge pull request #58125 from fire-forge/editor_property_overlap_fix

Fix EditorProperty icon overlapping text with checkbox

akien-mga committed:
Merge pull request #57904 from V-Sekai/fix_double_free_warning

Fixes some 'Attempted to free invalid ID' warnings in mobile renderer

akien-mga committed:
Merge pull request #57905 from V-Sekai/fix_instance_clear_warnings

Free surface vertex_buffers after vertex_arrays to silence warnings

timothyqiu committed:
Utilize mouse position when zooming with shortcuts in 2D editor

Faless committed:
Merge pull request #58111 from maksloboda/decoding-master-fix

Fixed variant decoding Segmentation Fault

akien-mga committed:
Merge pull request #58104 from Calinou/doc-audio-device

akien-mga committed:
Merge pull request #35679 from Calinou/doc-add-xml-schema

akien-mga committed:
Merge pull request #58081 from tavurth/bugfix/high-macos-cpu-usage

akien-mga committed:
Merge pull request #57877 from bruvzg/subpixel_gl_pos

akien-mga committed:
Merge pull request #57329 from Ev01/tileset-editor-minimum-zoom

Josephblt committed:
Added show and hide methods and updated doc API.

akien-mga committed:
Merge pull request #57863 from Calinou/editor-use-anchors-metadata-no-redundant

akien-mga committed:
Merge pull request #58136 from akien-mga/editor-includes-cleanup

Geometror committed:
Fix tooltip flickering in some cases

akien-mga committed:
Merge pull request #58144 from Josephblt/Godot-4-CanvasLayer-missing-show-hide

akien-mga committed:
Merge pull request #58147 from Geometror/fix-tooltip-flicker

akien-mga committed:
Use `switch` consistently in `_notification` (`scene` folder)

markdibarry committed:
Fix scroll_to_line ignoring line separation

akien-mga committed:
Merge pull request #58088 from reduz/add-particle-userdata

akien-mga committed:
Merge pull request #57931 from Calinou/sky-default-filter-realtime

akien-mga committed:
Merge pull request #58151 from akien-mga/notification-switch-scene

Sauermann committed:
Connect notifications from Container to Viewport
mouse enter+exit

KoBeWi committed:
Automatically select source of the picked tile

jmb462 committed:
Port existing _notification code to use switch statements  (part 1/3)

Calinou committed:
Fix default sky process mode not being Real-Time

The change in 09563e4bd8fa4dda8802b3b48f111f3c1de499d8 wasn't
consistently carried out everywhere needed.

Zylann committed:
Fix loading of binary resources with 64-bit floats

megalobyte committed:
Update if == NOTIFICATION_* to switch statements to match coding style

timothyqiu committed:
Add GridMap.get_used_cells_by_item

akien-mga committed:
Merge pull request #58170 from timothyqiu/gridmap-used-id

Add `GridMap.get_used_cells_by_item`

akien-mga committed:
Merge pull request #58165 from Calinou/fix-default-sky-process-mode

Fix default sky process mode not being Real-Time

timothyqiu committed:
Fix VisualScript crash when using Set Index node

akien-mga committed:
HTML5: Fix build after #52742

Fixes #58172.

timothyqiu committed:
Make VMap::find_nearest return -1 when empty

clayjohn committed:
Properly exclude lights that are marked as sky only

akien-mga committed:
Merge pull request #44716 from Calinou/os-add-get-processor-name

akien-mga committed:
Merge pull request #58174 from timothyqiu/vs-set-index-crash

akien-mga committed:
Merge pull request #58173 from akien-mga/js-build-fix-is_project_manager

clayjohn committed:
Use prefiltered radiance

akien-mga committed:
i18n: Sync editor translations with Weblate

Minor proofreading of new OSX export strings.

(cherry picked from commit 9f09251027a485f14740dce7b753e962d26bff78)

akien-mga committed:
Merge pull request #58159 from KoBeWi/source_jumper

akien-mga committed:
i18n: Sync classref translations with Weblate

(cherry picked from commit 9cb169504a5f529a542841bff8ff6cf91cd0b82d)

akien-mga committed:
Merge pull request #58175 from clayjohn/VULKAN-sky-only-bug

akien-mga committed:
Merge pull request #58176 from timothyqiu/find-nearest

akien-mga committed:
Style: Cleanup single-line blocks, semicolons, dead code

Remove currently unused implementation of TextureBasisU, could be re-added
later on if needed and ported.

akien-mga committed:
Merge pull request #58154 from markdibarry/fix_scroll_to_line

akien-mga committed:
Merge pull request #58055 from markdibarry/add_get_line_offset

BastiaanOlij committed:
We were only getting our available shader features, now we also enabling them

imjuniper committed:
Convert _notification methods to switch - Chunk C

akien-mga committed:
Merge pull request #58163 from jmb462/notification-chunk-A

fazil47 committed:
Default material is assigned to meshes without material for glTF export

MisoMosiSpy committed:
Added icons for missing types in visual script’s icon list.

akien-mga committed:
Merge pull request #58169 from megalobyte/update-notificatin-if-to-switch-batch-B

akien-mga committed:
Merge pull request #58182 from akien-mga/style-cleanup-if-semicolons-deadcode

akien-mga committed:
Merge pull request #58190 from MisoMosiSpy/vs_icons

timothyqiu committed:
Fix documentation about TreeItem button ID

poiati committed:
Fix extension registration order.

poiati committed:
Fix extension registration order.

akien-mga committed:
Merge pull request #58187 from jakobbouchard/notification-switch-chunk-c

Convert _notification methods to switch - Chunk C

Zylann committed:
Add missing flag when encode_variant writes math types with doubles

akien-mga committed:
Merge pull request #58197 from timothyqiu/tree-item-button-doc

Fix documentation about `TreeItem` button ID

Calinou committed:
Add a property hint for the Line2D Round Precision property

This prevents choosing extremely high values which cause performance
issues for no visual benefit.

timothyqiu committed:
Fix mismatched BBCode tags in classref

timothyqiu committed:
Translate link titles in classref XML

ellenhp committed:
Merge pull request #58130 from timothyqiu/ogg-seek-crash

Fix infinite recursion when seeking Ogg Vorbis file

akien-mga committed:
Merge pull request #58222 from timothyqiu/doctool-link-title-i18n

Translate link titles in classref XML

akien-mga committed:
Merge pull request #58219 from timothyqiu/bbcode-tag

Fix mismatched BBCode tags in classref

akien-mga committed:
Merge pull request #58212 from Calinou/line2d-round-precision-add-property-hint

Add a property hint for the Line2D Round Precision property

akien-mga committed:
Merge pull request #58189 from fazil47/gltf-export-default-material-fix

Default material is assigned to meshes without material for glTF export

akien-mga committed:
Merge pull request #56032 from diddykonga/get_children_of_type

bruvzg committed:
[Windows] Fix Vulkan driver crash on sub-window minimization.

YuriSizov committed:
Fix a typo in PROPERTY_HINT_ENUM_SUGGESTION description

akien-mga committed:
Merge pull request #58238 from pycbouh/docs-enum-suggestion-typo

timothyqiu committed:
Fix wording in AnimationNode classref

SaracenOne committed:
Return a correctly typed variant in case of a function error to prevent hard crashes

KoBeWi committed:
Connect text_submitted of built-in script name

akien-mga committed:
Merge pull request #58243 from timothyqiu/doc-ani-node

akien-mga committed:
Merge pull request #58177 from clayjohn/VULKAN-prefiltered-radiance

akien-mga committed:
Merge pull request #58246 from KoBeWi/submit_yer_name

Calinou committed:
Use versioned URL for "Online Documentation" buttons in the editor

This also avoids a redirect, making the page appear to load faster.

akien-mga committed:
Merge pull request #58249 from Calinou/editor-online-docs-use-versioned-url

akien-mga committed:
Merge pull request #58186 from BastiaanOlij/fix_shader_features_init

Enable features we require when creating a Vulkan Device

EMBYRDEV committed:
Updated out of date CharacterBody documentation.

clayjohn committed:
Revert "Fix default sky process mode not being Real-Time"

This reverts commit aa1cba1d53270cd14339141082d9d5dc9c3857ef.

clayjohn committed:
Revert "Default to real-time radiance map filter for environment sky reflections"

This reverts commit 09563e4bd8fa4dda8802b3b48f111f3c1de499d8.

akien-mga committed:
Merge pull request #58236 from bruvzg/win_min_fix

akien-mga committed:
Merge pull request #58196 from poiati/fix-extension-registration-order

akien-mga committed:
Merge pull request #58106 from Calinou/audioserver-capture-device-property

piiertho committed:
Fix default value for DisplayServer::virtual_keyboard_show Rect2 parameter

akien-mga committed:
Merge pull request #58268 from piiertho/bugfix/fix-display-server-virtual-keyboard-show-default-argument

akien-mga committed:
Merge pull request #58266 from clayjohn/revert-sky-change

akien-mga committed:
Merge pull request #58261 from STUDIOEMBYR/CharacterBody-docs-update

SaracenOne committed:
Make 'is_attribute' false during parse error to prevent crash

akien-mga committed:
Revert "Fix extension registration order."

This reverts commit 94ef200bab8feb5024b2c64ac14daa24e0c4cc55.

This broke extension loading.
Fixes #58273.

rsubtil committed:
Fix decompression functions not returning errcodes

akien-mga committed:
Merge pull request #58276 from akien-mga/revert-58196

akien-mga committed:
Merge pull request #54513 from EricEzaM/3d-editor-inputs-fix

akien-mga committed:
Merge pull request #58278 from Ev1lbl0w/fix_decompress_retcode

rburing committed:
In final phase of test_body_motion, move and cull AABB for body once, instead of for every shape

bruvzg committed:
[Editor] Fix "en" editor translation detection.

Calinou committed:
Document that meshes and lights must be fully set up before baking VoxelGI

akien-mga committed:
Merge pull request #58287 from rburing/cull_aabb_for_body_once

In final phase of test_body_motion, move and cull AABB for body once instead of for every shape

akien-mga committed:
Merge pull request #58291 from Calinou/doc-voxelgi-procedural-bake-deferred

Document that meshes and lights must be fully set up before baking VoxelGI

akien-mga committed:
Merge pull request #58288 from bruvzg/fix_editor_en_locale

[Editor] Fix "en" editor translation detection.

BastiaanOlij committed:
vkQueueSubmit needs pWaitDstStageMask sized to waitSemaphoreCount

timothyqiu committed:
Fix crash in Create New Node dialog with certain user-created scripts

timothyqiu committed:
Fix crash when importing small GLTF mesh

timothyqiu committed:
Fix crash when importing DAE model

akien-mga committed:
Merge pull request #58205 from Zylann/fix_variant_encode_with_doubles

Add missing flag when encode_variant writes math types with doubles

akien-mga committed:
Merge pull request #58166 from Zylann/fix_binary_resource_with_doubles

Fix loading of binary resources with 64-bit floats

akien-mga committed:
Merge pull request #58306 from timothyqiu/create-node

Fix crash in Create New Node dialog with certain user-created scripts

akien-mga committed:
Merge pull request #58307 from timothyqiu/normals-size

Fix crash when importing small GLTF mesh

akien-mga committed:
Merge pull request #58308 from timothyqiu/dae-for

Fix crash when importing DAE model

akien-mga committed:
Merge pull request #58042 from Sauermann/fix-viewport-border-notifications

Fix Viewport mouse enter+exit notifications

Sauermann committed:
Update variable name to reflect usage

akien-mga committed:
Merge pull request #58311 from Sauermann/fix-mouse-in-window-variable-name

Update variable name to reflect usage

KoBeWi committed:
Fix CanvasItem visibility propagation

naithar committed:
[iOS] Fix incorrect method name for input dragging event

akien-mga committed:
Merge pull request #58317 from naithar/fix/ios-drag-input

akien-mga committed:
Core: Use forward declares for Vector3/Vector3i

Add add Vector3 operator in Vector3i.

Sauermann committed:
Fix vertical icons in container sizing

megalobyte committed:
Don't process VisualScriptNodes

Powerbyte7 committed:
[HTML5] Add Stadia controller to database

Add web support for the Stadia controller.

akien-mga committed:
Merge pull request #58323 from Sauermann/fix-vertical-icons

Fix vertical icons in Container Sizing

akien-mga committed:
Merge pull request #57163 from winterpixelgames/feature/allow-disable-atlas-texture-alpha-trim

Allow disabling the alpha trim on texture atlas creation.

aaronfranke committed:
Show an error when setting a negative size on a collision shape

TokageItLab committed:
Fixed cubic interpolate with loop

timothyqiu committed:
Fix errors when updating script template list

akien-mga committed:
Merge pull request #58327 from megalobyte/fix-vs-call-crash

akien-mga committed:
Merge pull request #58321 from akien-mga/vector3-forward-declares

zombietype committed:
Fixing iteration for extension level loading.

Extensions are not getting instantiating properly due to iteration calling the wrong levels for loading.

akien-mga committed:
Merge pull request #58329 from Powerbyte7/patch-1

akien-mga committed:
Sync controller mappings DB with SDL2 community repo

Synced with gabomdq/SDL_GameControllerDB@94b76208bc07a62a63575bc16da99411d325aac8

timothyqiu committed:
Fix GLTF exporter crash when using GridMap

zombietype committed:
Resolving suggestions on comment formatting.

SaracenOne committed:
Fix nullptr error in _update_padded_texture on texture with no image

Sauermann committed:
simplify calculation of max in grayscale

akien-mga committed:
Merge pull request #58363 from Sauermann/fix-3max

KoBeWi committed:
Fix 2D instance drop position

Commit:
Fixed issue with Godot modifications to polypartition.cpp third-party file

akien-mga committed:
Merge pull request #58352 from timothyqiu/script-template

Fix errors when updating script template list

akien-mga committed:
Merge pull request #58372 from KoBeWi/orphan_ci

Fix 2D instance drop position

akien-mga committed:
Merge pull request #58351 from akien-mga/sync-gamecontrollerdb

Sauermann committed:
Update doc for match: empty argument doesn't match anything

akien-mga committed:
Merge pull request #58367 from V-Sekai/tileset_atlas_crash_fix

akien-mga committed:
Merge pull request #58275 from V-Sekai/attribute_parse_crash_fix

akien-mga committed:
Merge pull request #58315 from KoBeWi/fixibility

akien-mga committed:
Merge pull request #58350 from kidrigger/patch-1

mhilbrunner committed:
Merge pull request #58385 from Sauermann/fix-match-empty

Update doc for match: empty argument doesn't match anything

Sauermann committed:
Use bresenham_line() in EditorAtlasPacker::chart_pack

Faless committed:
[Net] Fix multi-peer path-only replication.

It used to check if a net_id was ever assigned to that node to detect
when to send the path confirm to the remote peer.
This is wrong, because the same net_id is shared for all the remote
peers, but sent one by one.
Instead we now check if it's either not assigned or if the assigned
net_id is a cache ID, and in that case ensure that the remote peer has
been notified.

This can be further improved by unifying the cache interface, but for
now it's a fast fix to get path-only sync to work.

Calinou committed:
Fix typos in editor font preload dialog warning messages

Faless committed:
[Net] Optimize object cache notification send for single peer.

It used to always cycle all the peers when checking which one needed to
be notified, now it only does that when the target is multiple (i.e.
broadcast or exclusion).

akien-mga committed:
Merge pull request #58397 from Calinou/editor-font-preload-fix-typo

Fix typos in editor font preload dialog warning messages

akien-mga committed:
Merge pull request #58304 from BastiaanOlij/fix_queue_submit

vkQueueSubmit needs pWaitDstStageMask sized to waitSemaphoreCount

akien-mga committed:
Merge pull request #58400 from Faless/mp/4.x_repl_path_stopgap

[Net] Fix multi-peer path-only replication, optimize single peer object cache.

akien-mga committed:
Merge pull request #58376 from novaplusplus/master

Fixed issue with godot's changes to polypartition third-party code

akien-mga committed:
Merge pull request #58399 from Sauermann/fix-bresenham-usage

Use bresenham_line() in EditorAtlasPacker::chart_pack

Commit:
Rename motion_velocity to velocity

SaracenOne committed:
Fixes cyclic detection from variables assigning themselves to themselves in autocomplete, and restricts initialization of variables from other variables which have not been declared above it in class body

clayjohn committed:
Use properly use non-perceptual roughness when filtering radiance

clayjohn committed:
Fix shader compilation error with anisotropy

akien-mga committed:
Merge pull request #58331 from poiati/fix-extension-registration-order-2

akien-mga committed:
Merge pull request #58418 from clayjohn/Vulkan-cubemap-proughness

akien-mga committed:
Merge pull request #58419 from clayjohn/Vulkan-anisotropy-crash

bruvzg committed:
[Windows] Fix borderless window flag toggle and restoring minimized borderless window.

akien-mga committed:
Merge pull request #58420 from bruvzg/fix_borderless_minimize

akien-mga committed:
Merge pull request #57538 from floppyhammer/fix-gpu-particles-2d-offset

akien-mga committed:
Merge pull request #58344 from TokageItLab/fix-cubic-interpolate-with-loop

akien-mga committed:
CI: Use mono_static=yes for Mono builds

This removes the dependency on shared libmonosgen installed locally
and makes the artifacts usable as standalone for testing without
needing a full Mono install.

akien-mga committed:
Merge pull request #58426 from akien-mga/ci-mono-static

akien-mga committed:
Merge pull request #58343 from aaronfranke/negative-shape-warning

novaplusplus committed:
Fix incorrect SurfaceTool example in class docs

novaplusplus committed:
Document String's humanize_size static method

lawnjelly committed:
Fix GIProbe gizmo out of bounds crash

The GIProbe gizmo was writing values in 3 dimensions to Vector2s. This error was previously being masked by the Vector2 accessor, but now results in a crash or ERR_FAIL message.

This PR removes the Vector2s as they were unused.

akien-mga committed:
Merge pull request #58442 from novaplusplus/docs-string-humanize-size

Document String's humanize_size static method

akien-mga committed:
Merge pull request #58440 from novaplusplus/novas-doc-stuff

Fix incorrect surfacetool example in class docs

novaplusplus committed:
Added or improved documentation to a few more String methods

akien-mga committed:
Merge pull request #57664 from skyace65/TriangulateInfo

Add more information on triangulate polygon

akien-mga committed:
Merge pull request #51716 from clayjohn/VULKAN-optimized-GGX

Use Filament specular model and parametrization

akien-mga committed:
Merge pull request #58411 from cbscribe/rename_motion_velocity

akien-mga committed:
Merge pull request #58415 from V-Sekai/cyclic_assignment_gdscript_fixes

akien-mga committed:
Merge pull request #57956 from Eoin-ONeill-Yokai/audio_stream_hotfix

akien-mga committed:
Merge pull request #58444 from lawnjelly/giprobe_gizmo_fix4

akien-mga committed:
Merge pull request #58354 from timothyqiu/gridmap-export

akien-mga committed:
Merge pull request #56394 from BastiaanOlij/OpenXR_Core4

akien-mga committed:
Merge pull request #58244 from V-Sekai/typed_fail_case_return

akien-mga committed:
Merge pull request #58134 from timothyqiu/shortcut-zoom-mouse-pos

akien-mga committed:
Merge pull request #58448 from novaplusplus/wow-more-string-docs

groud committed:
Fix llvm compilation by adding missing override

akien-mga committed:
Merge pull request #58472 from groud/fix_llvm_compilation

AlbertFay committed:
Merge branch 'text-under-scrollbar' of https://github.com/AlbertFay/godot into text-under-scrollbar
